ISHPullUp/ISHPullUpViewController.m

```diff
@@ -373,16 +378,77 @@ - (void)updateCachedHeightsWithSize:(CGSize)size  {
     self.maximumBottomHeightCached = [self maximumBottomHeightWithSize:size];
 }
 
+- (void)setBottomHidden:(BOOL)bottomHidden {
+    [self setBottomHidden:bottomHidden animated:NO];
+}
+
+- (void)setBottomHidden:(BOOL)bottomHidden animated:(BOOL)animated {
+    if (bottomHidden == _bottomHidden) {
+        return;
+    }
+
+    _bottomHidden = bottomHidden;
+    [self updateGestureEnabledState];
+
+    UIViewController *bottomVC = self.bottomViewController;
+
+    if (!bottomHidden) {
+        bottomVC.view.hidden = NO;
+        [bottomVC viewWillAppear:animated];
+    } else {
+        [bottomVC viewWillDisappear:animated];
+    }
+
+    void (^updateBlock)();
+    updateBlock = ^{
+        [self setDimmingViewHidden:[self dimmingViewShouldBeHidden] height:self.bottomHeight];
+        [self updateViewLayoutBottomHeight:self.bottomHeight withSize:self.view.bounds.size];
+    };
+
+    void (^completion)(BOOL);
+    completion = ^(BOOL finished) {
+        if (!finished) {
+            return;
+        }
+
+        if (bottomHidden) {
+            [bottomVC viewDidDisappear:animated];
```

Comment:
Should we rather use `weakSelf.bottomViewController` (get it once within the block, return early if `nil`) to ensure:

1. ...we get the current bottom vc
2. ...we don't hold on to the bottom vc if `self` has been released?